lib/api_loader.js

```diff
@@ -1,139 +1,9 @@
-      var id = prefixMap[match[1]], version = match[2];
-      if (serviceMap[id]) {
-        serviceMap[id].versions = serviceMap[id].versions || [];
-        if (serviceMap[id].versions.indexOf(version) < 0) {
-          serviceMap[id].versions.push(version);
-        }
-      }
-    }
-  });
-
-  Object.keys(serviceMap).forEach(function(identifier) {
-    serviceMap[identifier].versions = serviceMap[identifier].versions.sort();
-    serviceIdentifiers.push(identifier);
-    serviceNames.push(serviceMap[identifier].name);
-  });
-}
-
-function getServices() {
-  buildServiceMap();
-  return serviceIdentifiers;
-}
-
-function getServiceNames() {
-  buildServiceMap();
-  return serviceNames;
-}
-
-function serviceVersions(svc) {
-  buildServiceMap();
-  svc = serviceIdentifier(svc);
-  return serviceMap[svc] ? serviceMap[svc].versions : null;
-}
-
-function serviceName(svc) {
-  buildServiceMap();
-  svc = serviceIdentifier(svc);
-  return serviceMap[svc] ? serviceMap[svc].name : null;
-}
-
-function serviceFile(svc, version) {
-  buildServiceMap();
-  svc = serviceIdentifier(svc);
-  if (!serviceMap[svc]) return null;
-
-  var prefix = serviceMap[svc].prefix || svc;
-  var filePath;
-  ['min', 'api', 'normal'].some(function(testSuffix) {
-    filePath = apiRoot + '/' + prefix.toLowerCase() + '-' + version + '.' +
-           testSuffix + '.json';
-
-    return fs.existsSync(filePath);
-  });
-  return filePath;
-}
-
-function paginatorsFile(svc, version) {
-  buildServiceMap();
-  svc = serviceIdentifier(svc);
-  if (!serviceMap[svc]) return null;
-
-  var prefix = serviceMap[svc].prefix || svc;
-  return apiRoot + '/' + prefix + '-' + version + '.paginators.json';
-}
-
-function waitersFile(svc, version) {
-  buildServiceMap();
-  svc = serviceIdentifier(svc);
-  if (!serviceMap[svc]) return null;
-
-  var prefix = serviceMap[svc].prefix || svc;
-  return apiRoot + '/' + prefix + '-' + version + '.waiters2.json';
-}
-
-function load(svc, version) {
-  buildServiceMap();
-  svc = serviceIdentifier(svc);
-  if (version === 'latest') version = null;
-  version = version || serviceMap[svc].versions[serviceMap[svc].versions.length - 1];
-  if (!serviceMap[svc]) return null;
-
-  var api = require(serviceFile(svc, version));
-
-  // Try to load paginators
-  if (fs.existsSync(paginatorsFile(svc, version))) {
-    var paginators = require(paginatorsFile(svc, version));
-    api.paginators = paginators.pagination;
-  }
-
-  // Try to load waiters
-  if (fs.existsSync(waitersFile(svc, version))) {
-    var waiters = require(waitersFile(svc, version));
-    api.waiters = waiters.waiters;
-  }
-
-  return api;
-}
-
-function serviceIdentifier(svc) {
-  return svc.toLowerCase();
-}
-
-module.exports = {
-  serviceVersions: serviceVersions,
-  serviceName: serviceName,
-  serviceIdentifier: serviceIdentifier,
-  serviceFile: serviceFile,
-  load: load
+AWS.apiLoader = function(svc, version) {
+  return AWS.apiLoader.services[svc][version];
```

Comment:
Is it possible that the `AWS.apiLoader` may be called before the api is actually loaded and stored on `AWS.apiLoader.services`? If that happens, this would throw an error (eg "Cannot read property '11-30-2014' of undefined"). We could either throw a more helpful error (such as "S3 service not yet loaded. Require './clients/s3 first"), or better yet, require the file for them to load the service (since that's what the original `AWS.apiLoader` does for you anyway, and although I don't think this is a public method, adding that would allow us not to break this method).